doc/doxygen/headers/matrixfree.h

```diff
@@ -17,8 +17,91 @@
 /**
  * @defgroup matrixfree Matrix-free infrastructure
  *
- * This module describes the matrix-free infrastructure in deal.II. In
- * essence, the framework provided by the FEEvaluation class on top of the
+ * This module describes the matrix-free infrastructure in deal.II.
+ * An outline of how the primary groups of classes in deal.II interact with the
+ * matrix-free infrastructure is given by the following clickable graph,
+ * with a more detailed description below:
+ *
+ * @dot
+digraph G
+{
+  graph[rankdir="TB",bgcolor="transparent"];
+  node [fontname="FreeSans",fontsize=15,
+        shape=record,height=0.2,width=0.4,
+        color="black", fillcolor="white", style="filled"];
+  edge [color="black", weight=10];
+  subgraph base {
+    rank="same";
+  tria       [label="Triangulation",    URL="\ref grid", ];
+  fe         [label="FiniteElement",    URL="\ref feall"];
+  mapping    [label="Mapping",          URL="\ref mapping"];
+  quadrature [label="Quadrature",       URL="\ref Quadrature"];
+  }
+  dh         [label="DoFHandler",       URL="\ref dofs"];
+  subgraph linalglibs {
+    rank="same";
+    simd     [label="SIMD",     URL="\ref CUDAWrappers", fontname="FreeSans",fontsize=12,
```

Comment:
By the way, did you mean to give this a `\ref CUDAWrappers` even though it's the `SIMD` node? Or is this a copy-paste error?